services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -176,7 +177,7 @@ private void setupOfflineTokens(RealmModel realm, RealmRepresentation realmRep)
     }
 
     protected void createDefaultClientScopes(RealmModel realm) {
-        DefaultClientScopes.createDefaultClientScopes(session, realm, true);
+        EntityManagers.runInBatch(session, () -> DefaultClientScopes.createDefaultClientScopes(session, realm, true), false);
```

Comment:
Just out of curiosity, why do we need to run this in batch? It's just default client scopes, it's not that many.

Reply:
I see. Ok, let's keep it then.

Reply:
There were a couple of targeted changes like this that improved the performance of realm import in this pr. Users with hundreds of realms, who would choose to do an export as a single file, or during a migration will  see the same sort of problem with users with input - the persistence context grows too large, and things get progressively slower with each realm. 

There's a couple of existing issues about migration performance, which I'll follow-up on after this change is merged.